src/smtp_server.c

```diff
@@ -642,8 +662,57 @@ int smtp_hdlr_data(struct smtp_server_context *ctx, const char *cmd, const char
 	/* prepare response */
 	ctx->code = 354;
 	ctx->message = strdup("Go ahead");
-	ctx->node = smtp_cmd_lookup("BODY");
-	return SCHS_CHAIN;
+
+	smtp_server_response(stream, ctx->code, ctx->message);
```

Comment:
I believe you can simply call smtp_server_response(stream, 354, "Go ahead"). The message is a memory leak now anyway (because it's overwritten and the memory allocated by strdup() is not freed anywhere).